FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatController.swift

```diff
@@ -103,12 +103,14 @@ public final class HeartbeatController {
       )
     }
 
-    // Synchronously gets and returns the stored heartbeats, resetting storage
-    // using the given transform. If the operation throws, assume no
-    // heartbeat(s) were retrieved or set.
-    if let heartbeatsBundle = try? storage.getAndSet(using: resetTransform) {
```

Comment:
This was problematic because `getAndSet(using:)` returns an optional. In the latest Swift versions, the `try?` would do a "double-unwrap": it would convert the throwing result to an optional, unwrap it, and then unwrap the result of the API itself. It's pretty handy and cleans up the unwrapping logic. But, this behavior is not recognized on older Swift versions so i refactored things to be more backwards compatible. The new diff is perhaps easier to understand because the "double-unwrapping" behavior of `try?` may not be too obvious (I think i originally stumbled on it by mistake).

Reply:
LGTM